chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -512,6 +512,18 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             block_timestamp.as_duration_since_epoch() <= current_time + *max_future_offset,
             CheckBlockError::BlockFromTheFuture,
         );
+
+        if let GenBlockId::Block(prev_block_id) = header.prev_block_id().classify(self.chain_config)
+        {
+            let prev_block_index = self.get_block_index(&prev_block_id)?;
+            if let Some(prev_block_index) = prev_block_index {
+                ensure!(
+                    prev_block_index.status().is_valid(),
+                    CheckBlockError::InvalidParent(header.block_id())
+                );
+            }
```

Comment:
This check passes if the parent index is not found. Is that intentional? I guess the block ends up among orphans in that case. Is the valid parent check only performed once block is out of the orphan block set? I am not quite clear on the check sequence logic and the invariants upheld at each stage.

Reply:
Any changes to chainstate made here are superseded by the changes in https://github.com/mintlayer/mintlayer-core/pull/1019. I've updated the PR to indicate that. There's only a test left in the p2p now. We may not even want that because essentially we're testing chainstate. However, there may be some value in testing p2p -> chainstate integration.